pkgs/build-support/node/fetch-pnpm-deps/default.nix

```diff
@@ -0,0 +1,101 @@
+{ lib
+, stdenvNoCC
+, fetchurl
+, nodePackages
+, jq
+, moreutils
+, cacert
+, makeSetupHook
+}: let
+  # The names of these attributes refer to the pnpm lockfile version
+  pinnedVersions = {
+    "6_0" = {
+      name = "pnpm-8.15.5";
+      src = fetchurl {
+        url = "https://registry.npmjs.org/pnpm/-/pnpm-8.15.5.tgz";
+        sha512 = "sFGjLH5pWDO4SSbTspuMylclS1ifBknYmcbp0O22cLkex+KkNFm65zdZu1zmGcMmbxFr+THOItHvF1mn5Fqpbw==";
+      };
+    };
+    "7_0" = {
+      name = "pnpm-9.0.6";
+      src = fetchurl {
+        url = "https://registry.npmjs.org/pnpm/-/pnpm-9.0.6.tgz";
+        hash = "sha256-BiTjDv+GbN6zY7FQYb23/ZQlsXvBu0LCL19O/eoh9rM=";
+      };
+    };
+  };
+
+  pnpms = builtins.mapAttrs (_: value: nodePackages.pnpm.overrideAttrs value) pinnedVersions;
```

Comment:
Are these attributes useable for something other then `fetchPnpmDeps`? Perhaps they should be accessible in `all-packages` as well?

Reply:
They are *technically* accessible already using `fetchPnpmDeps_${version}.pnpm`. I don't think `nodePackages` supports versioned packages like this, though, correct me if I am wrong.